qcengine/programs/qchem.py

```diff
@@ -259,3 +263,371 @@ def parse_output(self, outfiles: Dict[str, str], input_model: "AtomicInput") ->
+                    cf = 1
+                else:
+                    cf = constants.conversion_factor("angstrom", "bohr")
+
+                geometry = [list(map(lambda x: float(x) * cf, line.split()[1:4])) for line in lines[1:]]
+                molecule = {
+                    "molecular_multiplicity": spin,
+                    "molecular_charge": charge,
+                    "geometry": geometry,
+                    "symbols": symbols,
+                }
+                input_dict["molecule"] = molecule
+
+            _excluded_rem = {
+                "input_bohr",
+                "mem_total",
+                "mem_static",
+            }  # parts of the rem normally written by build_input, and which do not affect results
+            for item in _excluded_rem:
+                if item in input_dict["keywords"]:
+                    input_dict["keywords"].pop(item)
+
+        try:
+            qcscr_result = self.parse_output(outfiles, AtomicInput(**input_dict)).dict()
+        except KeyError:
+            qcscr_result = {"properties": {}, "provenance": provenance_stamp(__name__), **input_dict}
+
+        mobj = re.search(r"This is a multi-thread run using ([0-9]+) threads", outtext)
+        if mobj and qcscr_result["provenance"].get("nthreads", None) is None:
+            qcscr_result["provenance"]["nthreads"] = int(mobj.group(1))
+
+        mobj = re.search(r"Total job time:\s*" + NUMBER + r"s\(wall\)", outtext)
+        if mobj and qcscr_result["provenance"].get("wall_time", None) is None:
+            qcscr_result["provenance"]["wall_time"] = float(mobj.group(1))
+
+        mobj = re.search(r"Archival summary:\s*\n[0-9]+\\[0-9+]\\([\w\.\-]+)\\", outtext)
+        if mobj and qcscr_result["provenance"].get("hostname", None) is None:
+            qcscr_result["provenance"]["hostname"] = mobj.group(1)
+
+        mobj = re.search(r"\n\s*SCF\s+energy in the final basis set =\s+" + NUMBER + r"\s*\n", outtext)
+        if mobj and qcscr_result["properties"].get("scf_total_energy", None) is None:
+            qcscr_result["properties"]["scf_total_energy"] = float(mobj.group(1))
+
+        mobj = re.search(r"\n\s*Nuclear Repulsion Energy =\s+" + NUMBER + r"\s+hartrees\s*\n", outtext)
+        if mobj and qcscr_result["properties"].get("nuclear_repulsion_energy", None) is None:
+            qcscr_result["properties"]["nuclear_repulsion_energy"] = float(mobj.group(1))
+
+        mobj = re.search(r"\n\s*There are\s+(\d+) alpha and\s+(\d+) beta electrons\s*\n", outtext)
+        if mobj and qcscr_result["properties"].get("calcinfo_nalpha", None) is None:
+            qcscr_result["properties"]["calcinfo_nalpha"] = int(mobj.group(1))
+        if mobj and qcscr_result["properties"].get("calcinfo_nbeta", None) is None:
+            qcscr_result["properties"]["calcinfo_nbeta"] = int(mobj.group(2))
+
+        mobj = re.search(r"\n\s*There are\s+\d+ shells and\s+(\d+) basis functions\s*\n", outtext)
```

Comment:
Same here, anything we don't currently parse in `parse_output` should be moved there I think.